lib/bios/seabios.robot

```diff
@@ -0,0 +1,305 @@
+
+Get Hidden Security Register Submenu Construction
+    [Arguments]    ${checkpoint}=Esc=Exit    ${lines_top}=1    ${lines_bot}=1    ${opt_only}="${FALSE}"
+    # In most cases, we need to strip two lines:
+    #    TOP:
+    #    Title line, such as:    Dasharo System Features
+    #    BOTTOM:
+    #    Help line, such as:    F9=Reset to Defaults    Esc=Exit
+    ${submenu}=    Get Menu Construction    ${checkpoint}    ${lines_top}    ${lines_bot}
+    # Handling of additional exceptions appearing in submenus:
+    #    1. Drop unselectable strings from Device Manager
+    Remove Values From List    ${submenu}    Devices List
+
+    IF    ${opt_only} == ${TRUE}
+        # Handling exceptions caused by some options splitting into multiple lines.
+        # For Dasharo System Features options, we can assume that each entry has
+        # either ">", or "[ ]", or "< >". For other edk2 menus, this is not always
+        # the case (yet?).
+        FOR    ${entry}    IN    @{submenu}
+            ${status}=    Check If Menu Line Is An Option    ${entry}
+            IF    ${status} != ${TRUE}
+                Remove Values From List    ${submenu}    ${entry}
+            END
+        END
+    END
+    RETURN    ${submenu}
+
+Get Hidden Flash Lockdown Submenu Construction
+    [Arguments]    ${checkpoint}=Esc=Exit    ${lines_top}=1    ${lines_bot}=1    ${opt_only}="${FALSE}"
+    # In most cases, we need to strip two lines:
+    #    TOP:
+    #    Title line, such as:    Dasharo System Features
+    #    BOTTOM:
+    #    Help line, such as:    F9=Reset to Defaults    Esc=Exit
+    ${submenu}=    Get Menu Construction    ${checkpoint}    ${lines_top}    ${lines_bot}
+    # Handling of additional exceptions appearing in submenus:
+    #    1. Drop unselectable strings from Device Manager
+    Remove Values From List    ${submenu}    Devices List
+
+    IF    ${opt_only} == ${TRUE}
+        # Handling exceptions caused by some options splitting into multiple lines.
+        # For Dasharo System Features options, we can assume that each entry has
+        # either ">", or "[ ]", or "< >". For other edk2 menus, this is not always
+        # the case (yet?).
+        FOR    ${entry}    IN    @{submenu}
+            ${status}=    Check If Menu Line Is An Option    ${entry}
+            IF    ${status} != ${TRUE}
+                Remove Values From List    ${submenu}    ${entry}
+            END
+        END
+    END
+    RETURN    ${submenu}
+
+Enter Menu From Snapshot
+    [Documentation]    Enter given Menu option
+    [Arguments]    ${menu}    ${option}
+    ${key}=    Extract Menu Key    ${menu}    ${option}
+    Write Bare Into Terminal    ${key}
+
+Enter Menu From Snapshot And Return Sortbootorder Construction
+    [Documentation]    Enter given sortbootorder Menu option and return construction
+    [Arguments]    ${menu}    ${option}
+    ${key}=    Extract Sortbootorder Menu Key    ${menu}    ${option}
+    Write Bare Into Terminal    ${key}
```

Comment:
Link to the same potential issue with PiKVM where `Write Bare Into Terminal ` is used. I can see in UEFI we have always this IF/ELSE. It can be resolved in a better way, but not necessarily in this PR.